### DISCUSSION TITLE
Applying gradient after time interval

### DISCUSSION THREAD

**Discussion by mdh23rct** [idaholab/moose#28553] [Q&A Modules: Phase field]
Dear all,
I am trying to implement a spatial gradient (DD_Scaling) after a time interval. Part of the code looks like this:
```
"[./Scaling_DD]
    type = DerivativeParsedMaterial
    property_name = 'DD_Scaling'
    material_property_names = 'DD_bl hbl1'
    expression =  (hbl1*DD_bl )'
    outputs = exodus
    derivative_order = 2
  [../]
  [./density]
    type = GenericFunctionMaterial
    material_property_names = 'DD_Scaling'
    prop_names = 'density'
    prop_values = 'if(t<150,0,DD_Scaling)'
    #outputs = exodus
  [../]"
```

However, GenericFunctionMaterial does not allow using a derivativeparsedmaterial into it. So what is the way to use a spatial gradient after a time interval?

I appreciate your response. Thanks.

**Comment by GiudGiud** [comment]
Hello

I think a parsedMaterial will do well for defining this 'if' condition, using the expression parameter

**Reply by mdh23rct** [reply]
@GiudGiud changed the code to 
```
[./Scaling_DD]
    type = DerivativeParsedMaterial
    property_name = 'DD_Scaling'
    material_property_names = 'DD_bl hbl1'
    expression =  (hbl1*DD_bl )'
    outputs = exodus
    derivative_order = 2
  [../]
[./density]
    type = ParsedMaterial
    property_name = 'density'
    material_property_names = 'DD_Scaling'
    expression = 'if(t<150,0,DD_Scaling)'
    #outputs = exodus
  [../]
```

However, now showing error:
```
*** ERROR ***
Invalid function
if(t<150,0,DD_Scaling)
DD_Scaling
in ParsedMaterialHelper.
Syntax error: Unknown identifier
```
Any solution for that?
Thanks.

**Reply by GiudGiud** [reply]
how old is your version of moose?

I think t is not recognized by the ParsedMaterial
You ll need to define t as a function, with a ParsedFunction with use_xyzt set to true for example

**Reply by mdh23rct** [reply]
@GiudGiud I did not update MOOSE recently. good news is solved the issue. We need ParsedFunction with FunctionAux. appreciate, cheers.
